packages/create-next-app/templates/index.ts

```diff
@@ -201,8 +201,8 @@ export const installTemplate = async ({
       ...packageJson.devDependencies,
       typescript: "^5",
       "@types/node": "^20",
-      "@types/react": "^18",
-      "@types/react-dom": "^18",
+      "@types/react": "npm:types-react@beta",
+      "@types/react-dom": "npm:types-react-dom@beta",
```

Comment:
You'd also have to add `resolutions` or `overrides` depending on the package manager since [`@types/*` packages don't use peer dependencies](https://github.com/microsoft/DefinitelyTyped-tools/issues/433)